### PR TITLE
Posts list: Use JS to Sentence case "Add new post" label

### LINKED ISSUE #40027: Posts: Programmatically change button labels to sentence case
Continued from D40003-code, partially addressed in #39692, and logged as a core issue in https://core.trac.wordpress.org/ticket/49616. Phew!

Where core will eventually adopt Sentence case for UI elements, we can use a JS solution in Calypso to temporarily change API-returned values (like the following one on the Posts page) rather than try to modify the API itself.

Screen Shot 2020-03-10 at 3 38 15 PM

"Add New Post" comes from the API and should read "Add new post".

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* Use JS to change "Add new post" label from Title Case to Sentence case to match changes made throughout Calypso in #40120 and #39692 

**Before**

Screen Shot 2020-03-18 at 2 36 05 PM

**After**

Screen Shot 2020-03-18 at 2 35 43 PM

#### Testing instructions

* Switch to this PR
* Navigate to /posts
* Note the "Add new post" button text.

Fixes #40027

### CODE REVIEW

**Comment by danhauk** [comment]
This is an interesting approach! I'm not sure it will work in all cases though. For example, in German, all nouns are capitalized everywhere in the sentence. So proper sentence-casing for German would be "Neue Beitrang erstallen". Making it all lowercase would probably look like a bug in translation.

It looks like the German translation in core has correct sentence casing (I'm not sure why English is Title Case):

image
Here, "Beitrang" is "post" as a noun

**Comment by sixhours** [comment]
Ahh, that makes total sense. I'm not sure we can do this on the client side for all languages in that case; @gwwar @mmtr should we just wait for core to catch up and leave this button alone for now? Or is there another way to change the case on the client side so it works for all languages?

**Comment by gwwar** [comment]
I would target this for only EN variants, then rip out once we've updated strings in core.

File: client/my-sites/post-type-list/index.jsx [hunk 1]
```diff
@@ -191,16 +191,25 @@ class PostTypeList extends Component {
...
-		const { editorUrl, postLabels } = this.props;
+		const { editorUrl, postLabels, localeSlug } = this.props;
 
 		if ( ! postLabels ) {
 			return null;
 		}
 
+		let addNewLabel = postLabels.add_new_item;
+
+		if ( 'en' === localeSlug ) {
```
Comment (gwwar) [inline client/my-sites/post-type-list/index.jsx]: Don't forget en-gb :D

File: client/my-sites/post-type-list/index.jsx [hunk 2]
```diff
@@ -191,16 +191,25 @@ class PostTypeList extends Component {
...
+		const { editorUrl, postLabels, localeSlug } = this.props;
 
 		if ( ! postLabels ) {
 			return null;
 		}
 
+		let addNewLabel = postLabels.add_new_item;
+
+		if ( 'en' === localeSlug || 'en-gb' === localeSlug ) {
+			// Temporary workaround to Sentence case label from core API for EN lang
```
Comment (mmtr) [inline client/my-sites/post-type-list/index.jsx]: Can we maybe include a link to the Core Track ticket for extra context?

Comment (sixhours) [reply client/my-sites/post-type-list/index.jsx]: Good idea! I'll do that with a TODO note so we know to remove it after that ships.

**Pull request merged**